### PR TITLE
Add spinner for async actions

### PR DESCRIPTION
**Pull request by lucaleivaloop**
#### :page_facing_up: Description:

Add spinner for async actions.

---

#### :heavy_check_mark: Tasks:

* Add spinner to `Button`.
* Removed unnecessary style.
* Add state where required.

---

#### :play_or_pause_button: Demo:
https://www.loom.com/share/60828db190024c818633b9dbf344823d

@loopstudio/react-devs

### CODE REVIEW

File: src/components/Form/Form.styles.js [hunk 1]
```diff
@@ -28,11 +28,8 @@ export const Button = styled.button`
   &:disabled {
     background-color: ${(props) => props.theme.color.indigo200};
     box-shadow: ${(props) => props.theme.boxShadow.none};
-
-    &:hover {
-      background-color: ${(props) => props.theme.color.indigo200};
-    }
   }
+
   ${(props) => props.styles};
```
Comment (Tiago-Zarzavidjian) [inline src/components/Form/Form.styles.js:33]: What is this for?

Comment (lucaleivaloop) [reply src/components/Form/Form.styles.js:33]: I didn't question it, it seems to be what you are describing though.

File: src/components/SignIn/Form.js [hunk 2]
```diff
@@ -24,10 +25,12 @@ const SignInForm = () => {
   const formMethods = useForm({ validationSchema });
 
   const onSubmit = async (values) => {
+    setIsLoading(true);
     try {
       await dispatch(signIn(values));
       history.replace('/');
     } catch (error) {
+      setIsLoading(false);
```
Comment (donatoaguirre24) [inline src/components/SignIn/Form.js:33]: What do you think about putting this in the `finally` block as you did above? (Only to keep consistency)

File: src/components/SignUp/Form.js [hunk 3]
```diff
@@ -28,10 +29,12 @@ const SignUpForm = () => {
   const formMethods = useForm({ validationSchema });
 
   const onSubmit = async (values) => {
+    setIsLoading(true);
     try {
       await dispatch(signUp({ locale: intl.locale, ...values }));
       history.replace('/');
     } catch (error) {
+      setIsLoading(false);
```
Comment (donatoaguirre24) [inline src/components/SignUp/Form.js:37]: Same

**Comment by mwolman** [comment]
The loading button doesn't have the same height as the original button (I'm seeing that in the video). Can you adjust it?

File: src/components/SignIn/Form.js (hunk 2 above)
Comment (lucaleivaloop) [reply src/components/SignIn/Form.js:33]: That would make it so we update the component after the `history.replace('/');` which would unmount the component if I'm not mistaken. Wouldn't it?

Comment (donatoaguirre24) [reply src/components/SignIn/Form.js:33]: Yep totally right! Didn't noticed 😅

File: src/components/Form/Form.styles.js (hunk 1 above)
Comment (Rminetti) [reply src/components/Form/Form.styles.js:33]: Yes, that is exactly what that prop is for

**Comment by emmaielle** [comment]
@mwolman we can include that style change in the button in a separate PR because Luca is lacking free time to work on this fix for now

**Comment by mwolman** [comment]
> @mwolman we can include that style change in the button in a separate PR because Luca is lacking free time to work on this fix for now

Or maybe another person can help 🤗

**Pull request merged**